flow360/component/geometry.py

```diff
@@ -208,7 +218,97 @@ def from_file(
         # For type hint only but proper fix is to fully abstract the Draft class too.
         return super().from_file(file_names, name, solver_version, length_unit, tags)
 
-    def _get_metadata(self):
-        # get the metadata when initializing the object (blocking)
-        # My next PR
-        pass
+    def _get_metadata(self) -> None:
+        "Blockingly trying to download the metadata and populate the metadata info"
+        start_time = time.time()
+        while self._webapi.status == Flow360Status.PROCESSED:
+            if time.time() - start_time > TIMEOUT_MINUTES * 60:
+                raise TimeoutError(
+                    "Timeout: Process did not finish within the specified timeout period"
+                )
+            time.sleep(10)
```

Comment:
1. Changed.
2. This is the tricky part, `is_final()` considers both UPLOADED and PROCESSED as final status. Geometry resource is flagged as UPLOADED instantly once file is uploaded. But the metadata will not be available until PROCESSED

Reply:
Maybe I add a funciton is_processed?

Reply:
see `status` property of Flow360Resource, it will call `is_final` so I think the line 224 will not update as expected. I think the best solution is to inherit from Flow360Status and create GeometryStatus

Reply:
I got error:
E   TypeError: GeometryStatus: cannot extend enumeration 'Flow360Status'